engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,6 +650,32 @@ protected void serviceStart() throws Exception
     }
   }
 
+  /**
+   * Modify the config object by adding SSL related parameters in a resource as the case may be so the WebApp uses
+   * correct SSL params
+   *
+   * @param config  Configuration to be modified
+   */
+  private void addSSLConfigResource(Configuration config)
+  {
+    SSLConfig sslConfig = dag.getValue(Context.DAGContext.SSL_CONFIG);
+    if (sslConfig != null) {
+      String nodeLocalConfig = sslConfig.getNodeLocalConfigPath();
+      if (StringUtils.isNotEmpty(nodeLocalConfig)) {
+        config.addResource(new Path(nodeLocalConfig));
+      } else {
+        // create a configuration object and add it as a resource
+        Configuration sslConfigResource = new Configuration(false);
+        String keyStorePath = sslConfig.getKeyStorePath();  // need to strip out the root
+        keyStorePath = (new File(keyStorePath)).getName();
```

Comment:
Actually this is the best I think. Coming up with a different way to derive the path would in fact be a repetition or redundancy. Note the flow of data as follows:

- user specifies keystorePath as "/home/user1/myKeystore.jks" on the client.
- StramClient/ApexCli copies this file into the app package as a "local resource" so it is made available on the app master side in the local file system in the app's directory as myKeystore.jks
- This code takes the value of keystorePath and takes only the filename part (getName()) and passes it on to Hadoop's WebAppUtils where the local resource is properly read using the name "myKeystore.jks" 

I don't see the need to do it differently.